library/init/algebra/ordered_group.lean

```diff
@@ -185,12 +185,23 @@ end ordered_cancel_comm_monoid
 
 class ordered_comm_group (α : Type u) extends add_comm_group α, partial_order α :=
 (add_le_add_left : ∀ a b : α, a ≤ b → ∀ c : α, c + a ≤ c + b)
-(add_lt_add_left : ∀ a b : α, a < b → ∀ c : α, c + a < c + b)
 
 section ordered_comm_group
 variable {α : Type u}
 variable [ordered_comm_group α]
 
+lemma ordered_comm_group.add_lt_add_left (a b : α) (h : a < b) (c : α) : c + a < c + b :=
+begin
+  rw lt_iff_le_not_le at h,
+  rw lt_iff_le_not_le,
+  split,
+  { apply ordered_comm_group.add_le_add_left _ _ h.1 },
+  { intro w,
+    let w : -c + (c + b) ≤ -c + (c + a) := ordered_comm_group.add_le_add_left _ _ w _,
```

Comment:
The original mathlib proof used `replace` instead of `let`, but I think they are equivalent here.

Reply:
I won't complain as long as the proof works.

Reply:
I thought `replace` came from mathlib. If it's in the lean core, I can use it too

Reply:
Oh, you're right. `replace` is indeed a mathlib tactic. You could use `have` there instead of `let`. There will be two variables named `w`  but Lean is smart enough here to figure out which one goes where. If it bothers you you can rename one of them to `w'`.